### PR TITLE
Feat/single video history

### PR DESCRIPTION
**Pull request by bhavyaj12**
### Added Single Video page and History 

- Modified user data context+reducer for historyVideos
- Installed React Player
- Added single video page and get single video utility
- Added history service utility functions
- History page with option to delete video and clear history
- Single video page with option to like, watch later, add to playlist
- Some routing modifications to redirect user to correct "from" location
 
**Please check app is RESPONSIVE and with theme change functionality**
**Ignore console logs, I'll remove them later**

### What not to check
- Search
- Category filter
- User profile

image
image
image

### CODE REVIEW

**Review by sashaboi — APPROVED** [review]
I didnt know you were such a gamer girl ! 
Code is very readable and review friendly, 
Liked the folder structure , it clearly shows how the data is flowing.
Looks good to merge!

File: src/pages/SingleVideoPage/SingleVideoPage.jsx [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+            height={"70vh"}
+            width={"80vw"}
+            onStart={recordHistoryHandler}
+          />
+          <p className="h3 single-video-title my-4">{singleVideo.title}</p>
+          <div className="single-video-btns my-2">
+            {/* <button className="single-video-action">
+              <ThumbUpOutlinedIcon />
+              Like
+            </button> */}
```
Comment (haarsh24) [inline src/pages/SingleVideoPage/SingleVideoPage.jsx:120]: remove commented code if not necessary

File: src/pages/SingleVideoPage/SingleVideoPage.jsx [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+                <span className="mx-2">
+                  <ThumbUpOutlinedIcon />
+                </span>
+                Like
+              </button>
+            )}
+            {/* <button className="single-video-action mx-2">
+              <WatchLaterOutlined />
+              Watch later
+            </button> */}
```
Comment (haarsh24) [inline src/pages/SingleVideoPage/SingleVideoPage.jsx:145]: same here

File: src/utilities/services/historyService.js [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+  try {
+    const {
+      data: { history },
+    } = await axios.delete(
+      `/api/user/history/all`,
+      {
+        headers: { authorization: token },
+      }
+    );
+    console.log("history after deleting all history", history);
```
Comment (haarsh24) [inline src/utilities/services/historyService.js:62]: remove console logs

File: src/utilities/services/historyService.js [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+    userDataDispatch({ type: "UPDATE_HISTORY", payload: history });
+    showToast("success", "Cleared history successfully.");
+  } catch (error) {
+    console.log(error);
```
Comment (haarsh24) [inline src/utilities/services/historyService.js:66]: here also

File: src/utilities/services/fetchUserData.js [hunk 5]
```diff
@@ -47,4 +46,19 @@ const getUserPlaylists = async (token, userDataDispatch) => {
...
+  const { showToast } = useToast();
+  try {
+    const {
+      data: { history },
+    } = await axios.get(`/api/user/history`, {
+      headers: { authorization: token },
+    });
+    userDataDispatch({ type: "UPDATE_HISTORY", payload: history });
+  } catch (error) {
+    console.log(error.message);
```
Comment (haarsh24) [inline src/utilities/services/fetchUserData.js:59]: same here

**Review by haarsh24 — APPROVED** [review]
Clean code,
Good folder structure,
Good UI,
Try to remove console logs and commented code if not necessary

**Pull request merged**